src/http/vault.ts

```diff
@@ -67,13 +69,13 @@ export class VaultClient extends JsonRpcClient {
         await this.post("request_replace", [request.toHex()]);
     }
 
-    async registerVault(collateral: string, hash: string): Promise<void> {
-        const btcAddress = new this.constr["H160"](this.registry, hash);
+    async registerVault(collateral: string): Promise<BtcAddress> {
         const request = new this.constr["RegisterVaultJsonRpcRequest"](this.registry, {
             collateral: new BN(collateral),
-            btc_address: btcAddress,
         });
-        await this.post("register_vault", [request.toHex()]);
+        let response = await this.post("register_vault", [request.toHex()]);
+        const result = new this.constr["RegisterVaultJsonRpcResponse"](this.registry, response.result);
+        return result.address;
```

Comment:
When interacting with the UI, best is to use strings and not parachain types so we keep the conversion in the lib.